library_generation/post_processing/postprocessing_utilities.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+#
+# Main functions to interact with owlbot post-processor and postprocessing
+# scripts
+
+# returns the metadata json path if given, or defaults to the one found in
+# $repository_path
+# Arguments
+# 1 - repository_path: path from output_folder to the location of the library
+# containing .repo-metadata. It assumes the existence of google-cloud-java in
+# the output folder
+# 2 - output_folder: root for the generated libraries, used in conjunction with
+get_repo_metadata_json() {
+  local repository_path=$1
+  local output_folder=$2
+  >&2 echo 'Attempting to obtain .repo-metadata.json from repository_path'
+  local default_metadata_json_path="${output_folder}/${repository_path}/.repo-metadata.json"
+  if [ -f "${default_metadata_json_path}" ]; then
+    echo "${default_metadata_json_path}"
+  else
+    >&2 echo 'failed to obtain json from repository_path'
+    exit 1
+  fi
+}
+
+# returns the owlbot image sha contained in google-cloud-java. This is default
+# behavior that may be overriden by a custom value in the future.
+# Arguments
+# 1 - output_folder: root for the generated libraries, used in conjunction with
+# 2 - repository_root: usually "google-cloud-java". The .OwlBot.yaml
+# file is looked into its .github folder
+get_owlbot_sha() {
+  local output_folder=$1
+  local repository_root=$2
+  if [ ! -d "${output_folder}/${repository_root}" ];
+  then
+    >&2 echo 'No repository to infer owlbot_sha was provided. This is necessary for post-processing' >&2
+    exit 1
+  fi
+  >&2 echo "Attempting to obtain owlbot_sha from monorepo folder"
+  owlbot_sha=$(grep 'sha256' "${output_folder}/${repository_root}/.github/.OwlBot.lock.yaml" | cut -d: -f3)
+  echo "${owlbot_sha}"
+}
+
+# Runs the owlbot post-processor docker image.
+# Arguments
+# 1 - workspace: the location of the grpc,proto and gapic libraries to be
+# processed
+# owlbot
+# 4 - scripts_root: location of the generation scripts
+# 5 - destination_path: used to transfer the raw grpc, proto and gapic libraries
+# 6 - repository_path: path from output_folder to the location of the source of
```

Comment:
I still don't fully understand the usefulness of `repository_path`? If it's for the poms, they should be already in the output folders, am I missing something?

Reply:
@blakeli0 the pre-existing poms with their folder structure are indeed necessary for the postprocessor. 

We need `google-cloud-java` to be present beforehand in `output_folder` as well, but the script does not know which folder in the monorepo should obtain the poms, `owlbot.py` and `.OwlBot.yaml` from. I don't think there is a good heuristic to infer the monorepo path from `proto_path` or `destination_path`. For example looking for `.OwlBot.yaml` files that have a `deep-copy` containing `destination_path` may work but we have cases where the owlbot files do not deal with a single gapic library.

To clarify, the purpose of `repository_path` is to give the script a way to find the monorepo folder associated with the library being generated. This folder contains the necessary files for owlbot copy and postprocessor.

Reply:
As discussed, I removed usage of `destination_path`. The integration test, however, will pass `repository_path` (e.g. java-asset) as `destination_path`. 

NOTE: The usage of owlbot copy works with a yaml file [expecting](https://github.com/googleapis/google-cloud-java/blob/474eb74f8dd6d0c7abf480c5212da44a006272bb/java-asset/.OwlBot.yaml#L28) a certain input folder structure. For example, that yaml has a `deep-copy` section with wildcards in order to build the `owl-bot-staging` folder (i.e. owlbot copy sends the files stored in `deep-copy` to `owlbot-staging` and the owlbot postprocessor runs on them)
I'm taking advantage of the wildcards for owlbot copy to work. The path in this deep-copy section is referring a path in googleapis-gen ([example](https://github.com/googleapis/googleapis-gen/tree/master/google/cloud/asset/v1)) which is could be something like `google-cloud-asset-v1-java` but I just named it `generated-java` (which will be ok for the wildcard)

Reply:
I'm yet to update the readme